src/gd.c

```diff
@@ -2201,9 +2201,9 @@ static void _gdImageFillTiled(gdImagePtr im, int x, int y, int nc)
 	FILL_PUSH(y+1, x, x, -1);
 	while (sp>stack) {
 		FILL_POP(y, x1, x2, dy);
-		for (x=x1; x>=0 && (!pts[y + x*wx2] && gdImageGetPixel(im,x,y)==oc); x--) {
```

Comment:
pts is a flattened array of pixels.  it would probably be easier to understand if it were used as a 2d array.  but it's not (currently).
```pts = (char *) gdCalloc(im->sy * im->sx, sizeof(char));```

so when the code here is trying to do `pts[(x,y)]`, it does so by `pts[x * im->sx + y]`.  changing it to `pts[x * wy2 + y]` no longer properly indexes the array.  it might not violate the bounds, but if you changed it to `pts[0]` everywhere, that too would not violate the bounds :).